### PR TITLE
Refactor store and use sync.map as data store

### PR DESCRIPTION
**Pull request by MartinForReal**
Signed-off-by: MartinForReal <fanshangxiang@gmail.com>

#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
remove k/client-go/tools/cache dependency and provide a more performant cache.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/provider/azure_vmss_cache.go [hunk 1]
```diff
@@ -153,28 +153,6 @@ func (ss *ScaleSet) newVMSSVirtualMachinesCache(resourceGroupName, vmssName, cac
 
 	getter := func(key string) (interface{}, error) {
 		localCache := &sync.Map{} // [nodeName]*vmssVirtualMachinesEntry
-
-		oldCache := make(map[string]vmssVirtualMachinesEntry)
-
-		if vmssCache, ok := ss.vmssVMCache.Load(cacheKey); ok {
```
Comment (andyzhangx) [inline pkg/provider/azure_vmss_cache.go:159]: why remove this code?

Comment (MartinForReal) [reply pkg/provider/azure_vmss_cache.go:159]: the change is reverted.

**Review by feiskyer — COMMENTED** [review]
/lgtm

**Pull request merged**